### PR TITLE
update to java vertx 4.1.2

### PR DESCRIPTION
**Pull request by Ako-Njang**
Added support for vertx 4 and updated test cases to accordingly.

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@
     <netty.version>4.1.52.Final</netty.version>
     <gson.version>2.2.4</gson.version>
     <jackson.version>2.11.3</jackson.version>
-    <vertx.version>3.9.5</vertx.version>
+    <vertx.version>4.1.2</vertx.version>
```
Comment (gaol) [inline java/pom.xml:38]: I think it is better to upgrade to the latest version of Vertx and its declared dependencies:
* Vertx: `4.2.6`
* Netty: `4.1.74.Final`
* Jackson: `2.13.1`

File: java/src/test/java/io/vertx/eventbusclient/test/TcpBusTest.java [hunk 2]
```diff
@@ -195,12 +195,12 @@ public void handle(Message<Object> event) {
       }
     }));
     vertx.eventBus().consumer("send_to_client", msg -> {
-      vertx.eventBus().send("client_addr", new JsonObject().put("message", "hello"));
+      vertx.eventBus().request("client_addr", new JsonObject().put("message", "hello"));
```
Comment (gaol) [inline java/src/test/java/io/vertx/eventbusclient/test/TcpBusTest.java:198]: This does not require a reply, so `send` is good.

File: java/src/test/java/io/vertx/eventbusclient/test/WebSocketBusTest.java [hunk 3]
```diff
@@ -65,13 +65,13 @@ protected void setUpBridges(TestContext ctx) {
     Router bridgeRouter = ebHandler.bridge(opts);
     router.mountSubRouter("/eventbus-test/", bridgeRouter);
     HttpServer server = vertx.createHttpServer(new HttpServerOptions().setMaxWebSocketFrameSize(MAX_WEBSOCKET_FRAME_SIZE).setMaxWebSocketMessageSize(MAX_WEBSOCKET_FRAME_SIZE))
-      .requestHandler(router::accept)
+      .requestHandler(router::handle)
```
Comment (gaol) [inline java/src/test/java/io/vertx/eventbusclient/test/WebSocketBusTest.java:68]: we can simply use `.requestHandler(router)`

File: java/src/test/java/io/vertx/eventbusclient/test/WebSocketBusTest.java [hunk 4]
```diff
@@ -65,13 +65,13 @@ protected void setUpBridges(TestContext ctx) {
...
       .listen(7000, ctx.asyncAssertSuccess());
 
     vertx.createHttpServer(new HttpServerOptions().setMaxWebSocketFrameSize(MAX_WEBSOCKET_FRAME_SIZE).setMaxWebSocketMessageSize(MAX_WEBSOCKET_FRAME_SIZE).setSsl(true).setKeyStoreOptions(
       new JksOptions().setPath("server-keystore.jks").setPassword("wibble")
     ))
-      .requestHandler(router::accept)
+      .requestHandler(router::handle)
```
Comment (gaol) [inline java/src/test/java/io/vertx/eventbusclient/test/WebSocketBusTest.java:74]: we can simply use `.requestHandler(router)`

**Review by gaol — CHANGES_REQUESTED** [review]

**Comment by Ako-Njang** [comment]
@gaol  thanks for the feedback I am done with the changes.

**Review by gaol — APPROVED** [review]

**Pull request merged**